sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeDisableAutomaticCheckForUpdateTest.java

```diff
@@ -61,4 +62,65 @@ public void manualCheckWhileAutomaticCheckDisabledChecksForUpdate() {
     }
+
+    @Test
+    public void manualCheckAfterActivityResumedWhileAutomaticCheckDisabledChecksForUpdate() {
+
+        /* Start then call disable automatic check for update after Distribute has started. */
+        Distribute.disableAutomaticCheckForUpdate();
+        start();
+        verify(mHttpClient, never()).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+        Distribute.getInstance().onActivityResumed(mActivity);
+        Distribute.checkForUpdate();
+
+        /* HTTP call done. */
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+    }
+
+    @Test
+    public void checkForUpdateAfterManualCheckCompletesDoesNotCheckForUpdatesAgain() {
+
+        /* Start in public mode with automatic check for updates disabled. */
+        Distribute.disableAutomaticCheckForUpdate();
+        start();
+        Distribute.getInstance().onActivityResumed(mActivity);
+        Distribute.checkForUpdate();
+
+        /* Check http call done. */
+        ArgumentCaptor<ServiceCallback> httpCallback = ArgumentCaptor.forClass(ServiceCallback.class);
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), httpCallback.capture());
+
+        /* Complete call with no new release (this will return the default mock mReleaseDetails with version 0). */
+        httpCallback.getValue().onCallSucceeded(mock(HttpResponse.class));
+
+        /* Restart and check for updates again. */
+        restartResumeLauncher(mActivity);
+
+        /* Do not call again. */
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), notNull(ServiceCallback.class));
+    }
+
+    @Test
+    public void manualCheckForUpdateAfterManualCheckCompletesChecksForUpdateAgain() {
+
+        /* Start in public mode with automatic check for updates disabled. */
+        Distribute.disableAutomaticCheckForUpdate();
+        start();
+        Distribute.getInstance().onActivityResumed(mActivity);
+        Distribute.checkForUpdate();
+
+        /* Check http call done. */
+        ArgumentCaptor<ServiceCallback> httpCallback = ArgumentCaptor.forClass(ServiceCallback.class);
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), httpCallback.capture());
+
+        /* Complete call with no new release (this will return the default mock mReleaseDetails with version 0). */
+        httpCallback.getValue().onCallSucceeded(mock(HttpResponse.class));
+
+        /* Resume the activity again and manually check for updates. */
+        Distribute.getInstance().onActivityResumed(mActivity);
```

Comment:
The activity is already in foreground in this test (line 109), this line is unnecessary.